pkg/ccl/streamingccl/streamproducer/stream_lifetime.go

```diff
@@ -311,6 +311,23 @@ func buildReplicationStreamSpec(
 		return nil, err
 	}
 
+	const repartitionFactor, repartitionedMin = 4, 16
```

Comment:
you mentioned to me that you came up with the number 4 for the `repartitionFactor` here based on the `InitialScanParallelism` somehow. Currently the initial scan parallelism is set to 16... how did you come up with 4?